src/components/MessagingView.tsx

```diff
@@ -4,333 +4,943 @@ import {FhirClientContext, FhirClientContextType} from "../FhirClientContext";
+                  if (MessagingView.isValidDate(event.target.value)) {
+                    this.setState({
+                      activeMessage: {
+                        ...this.state.activeMessage,
+                        date: event.target.value
+                          ? new Date(event.target.value).toISOString()
+                          : null,
+                      },
+                    });
+                  }
+                }}
+              />
+            )}
+            onChange={(newValue: Date | null) => {
+              this.setState({
+                activeMessage: {
+                  ...this.state.activeMessage,
+                  date: newValue?.toISOString(),
+                },
+              });
+            }}
+            disableFuture
+          ></DateTimePicker>
+        </Stack>
+        <TextField
+          multiline
+          rows={4}
+          fullWidth
+        //   label={`Message ${
+        //     this.state.activeMessage?.status
+        //   } at ${this.state.activeMessage?.date}`}
+        //   InputLabelProps={{ shrink: true, sx: {
+        //     color: "text.primary"
+        //   } }}
+          placeholder="Enter info on email/phone/letter contact"
+          value={this.state.activeMessage?.content ?? ""}
+          onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+            this.setState({
+              activeMessage: {
+                ...this.state.activeMessage,
+                content: event.target.value,
+              },
+            });
+          }}
+          sx={{
+            backgroundColor: "#FFF"
+          }}
+        ></TextField>
+      </>
+    );
+  }
+
+  private _buildNonSMSEntryComponent(): React.ReactNode {
+    const activeType = this.state.activeMessage.type;
+    return (
+      <>
+        <Stack direction={"column"} spacing={1}>
+          {activeType === "comment" && this._buildNoteEntryComponent()}
+          {activeType !== "comment" && this._buildManualMessageEntryComponent()}
+        </Stack>
+        <Box sx={{ marginTop: 1, textAlign: "right" }}>
+          <LoadingButton
+            variant="contained"
+            onClick={() => {
+              this.saveNonSMSMessage();
+            }}
+            loading={this.state.saveLoading}
+            disabled={!this._hasMessageContent() || !this._hasMessageDate()}
+          >
+            Save
+          </LoadingButton>
+        </Box>
+      </>
+    );
+  }
+
+  private saveNonSMSMessage() {
+    // @ts-ignore
+    let context: FhirClientContextType = this.context;
+    this.setState({ saveLoading: true });
+    const sentDate =
+      this.state.activeMessage?.status === "sent"
+        ? this.state.activeMessage.date
+        : null;
+    const receivedDate =
+      this.state.activeMessage?.status === "received"
+        ? this.state.activeMessage.date
+        : null;
+    const noteAboutCommunication = `Non-SMS ${this.state.activeMessage?.type} message, staff entered`;
```

Comment:
1. spelling fix
```suggestion
    const noteAboutCommunication = `Non-SMS ${this.state.activeMessage?.type} message, staff-entered`;
```
2. in what sense is this a 'message' if it is a comment? This is misleading. Users will think these are messages exchanged with the patient

Reply:
addressed. thanks.  I removed the text `message`.